docs/compose-file.md

```diff
@@ -163,44 +163,29 @@ accessible to linked services. Only the internal port can be specified.
 Extend another service, in the current file or another, optionally overriding
 configuration.
 
-Here's a simple example. Suppose we have 2 files - **common.yml** and
-**development.yml**. We can use `extends` to define a service in
-**development.yml** which uses configuration defined in **common.yml**:
+You can use `extends` on any service together with other configuration keys.
```

Comment:
You can use `extends` on any service together with other configuration keys.
The `extends` value must be a dictionary defined with a required `service` and an optional `file` key.

    extends:
      file: common.yml
      service: webapp

The `service` is what is being extended, , for example `web` or `database`. the
`file` is the location of a Compose configuration file defining that service. If
you omit the `file` Compose looks for the service configuration in the current
file. The `file` value can be an absolute or relative path. If you specify a
relative path, Compose treats it as relative to the location of the current
file. If you don't specify a `file`, Compose looks in the current configuration
file.

You can extend a service that itself extends another. You can extend
indefinitely. Compose does not support circular references and `docker-compose`
returns an error if it encounters one.